### DISCUSSION TITLE
Request: Add the ability to set filters while Dashboard is loading

### DISCUSSION THREAD

**Discussion by jmcapra** [apache/superset#24049] [Ideas]
Currently using version 2.0.0

About my current use case: I regularly need to load a large dashboard with many filters - I'll generally have a set of values ready in the clipboard for one of the filters, however the UI selector box does not become available until the pre-filters have loaded for ALL of the filters - this is understandable if I were wanting to select a value from the dropdown, however in my use case it leads to a lot of workflow friction. 

Ideally, I should be able to specify the filters I require BEFORE the dashboard/chart starts the query and puts unnecessary load on the DB. 

Similarly, for Charts, upon loading the page the query begins - if I update the filters, I need to press "Stop", then wait several seconds before I can begin "Update Chart" to execute the new query. The functionality here should mirror that in the dashboards - an "Apply Filters" button to update the chart regardless of whether or not it is already executing. Ideally, both Charts and Dashboards should have both the "Apply Filters" (if filters have changed), as well as a "Stop" button (if query is running). 

There should be an option not to automatically run charts upon opening, with a "Run" button for initial execution. 

Thoughts?

**Comment by ftwegner** [comment]
You can pre-set values of the filters when you follow the steps in the article [Apache Superset — Use URL to filter dashboards](https://www.blef.fr/superset-filters-in-url/). I guess this could help you. All you need to do afterwards is click "APPLY FILTERS". See [here](https://stackoverflow.com/questions/78559604/how-to-apply-a-url-filter-in-a-superset-dashboard) for details on the latter.

**Comment by Rafeethu** [comment]
It's better to have this option which prevents the initial data loading when the chart/ dashboard is loaded and user need to fill the filters and apply the filter to load the data.

The reason behind this is even if we set a default filter value, the database query is executed with those default values which might not necessary at all and reduce the report initial load time drastically.

**Reply by mbardolia** [reply]
I agree with @Rafeethu. Has there been any progress on this conversation/idea?
